test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -811,6 +811,36 @@ class ExportsTest {
     assertSame(constr, js.constructorOf[SJSDefinedTopLevelExportedClass])
   }
 
+  @Test def toplevel_exports_for_abstract_JS_classes_issue4117(): Unit = {
+    val constr =
+      if (isNoModule) global.TopLevelExportedAbstractJSClass
+      else exportsNamespace.TopLevelExportedAbstractJSClass
+
+    assertEquals("function", js.typeOf(constr))
+
+    val subclassFun = new js.Function("constr",
+       """
+       class SubClass extends constr {
```

Comment:
This needs an `assumeTrue(<es6>)`, no? I'm not sure it's a good idea for the test suite to rely on things we happen to provide.

Reply:
Good point. I have adapted the test so that it uses a function+prototype when we are not assuming ES2015.